File: inventory-management/src/main/java/com/inventory_management/service/SupplierService.java
package com.inventory_management.service;

import com.inventory_management.dto.request.SupplierRequestDTO;
import com.inventory_management.dto.response.SupplierResponseDTO;
import org.springframework.data.domain.Page;

public interface SupplierService {

    SupplierResponseDTO createSupplier(
            SupplierRequestDTO request
    );

    Page<SupplierResponseDTO> getAllSuppliers(
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    Page<SupplierResponseDTO> searchSuppliers(
            String keyword,
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    SupplierResponseDTO getSupplierById(
            Integer supplierId
    );

    SupplierResponseDTO updateSupplier(
            Integer supplierId,
            SupplierRequestDTO request
    );

    void deleteSupplier(
            Integer supplierId
    );

    void restoreSupplier(
            Integer supplierId
    );

}

File: inventory-management-frontend/src/services/supplierService.js
import axiosInstance from "../api/axiosConfig";

const supplierService = {

    /*
        Used by Product Form dropdown
        Returns active suppliers only
    */
    async getSuppliers() {

        const response = await axiosInstance.get(
            "/suppliers",
            {
                params: {
                    active: true,
                    page: 0,
                    size: 100,
                    sortBy: "supplierId",
                    sortDir: "asc"
                }
            }
        );

        return response.data.content;

    },



    async getActiveSuppliers(params) {

        const response = await axiosInstance.get(
            "/suppliers",
            {
                params: {
                    ...params,
                    active: true
                }
            }
        );

        return response.data;

    },



    async getInactiveSuppliers(params) {

        const response = await axiosInstance.get(
            "/suppliers",
            {
                params: {
                    ...params,
                    active: false
                }
            }
        );

        return response.data;

    },



    async searchSuppliers(params) {

        const response = await axiosInstance.get(
            "/suppliers/search",
            {
                params
            }
        );

        return response.data;

    },



    async createSupplier(data) {

        const response = await axiosInstance.post(
            "/suppliers",
            data
        );

        return response.data;

    },



    async updateSupplier(id, data) {

        const response = await axiosInstance.put(
            `/suppliers/${id}`,
            data
        );

        return response.data;

    },



    async deleteSupplier(id) {

        await axiosInstance.delete(
            `/suppliers/${id}`
        );

    },



    async restoreSupplier(id) {

        await axiosInstance.patch(
            `/suppliers/${id}/restore`
        );

    }

};

export default supplierService;